### PR TITLE
NOISSUE: Architectural supervision fixes

### PR DESCRIPTION
**Pull request by dgovorukhin**
Fixed all comments from https://docs.google.com/document/d/1LEGQZtAzt-Ii3JOiSlLWyiHR08UaMAjbxp1xmooyIa4/edit#

### CODE REVIEW

File: ledger-core/virtual/object/object.go [hunk 1]
```diff
@@ -489,3 +474,33 @@ func (sm *SMObject) checkPendingCounters(logger smachine.Logger) {
 		})
 	}
 }
+
+func (sm *SMObject) sharedAndPublishStateReport(
```
Comment (cyraxred) [inline ledger-core/virtual/object/object.go:478]: this method is better to return "error"

File: ledger-core/virtual/object/object.go [hunk 2]
```diff
@@ -489,3 +474,33 @@ func (sm *SMObject) checkPendingCounters(logger smachine.Logger) {
...
+	if !ctx.Publish(finalizedstate.BuildReportKey(sm.Reference), sdlStateReport) {
+		return ctx.Error(throw.New("failed to publish state report", struct {
+			Reference reference.Holder
+		}{sm.Reference}))
+	}
+
+	return smachine.StateUpdate{}
+}
+
+func (sm *SMObject) sharedAndPublishSMCallSummarySyncLink(
```
Comment (cyraxred) [inline ledger-core/virtual/object/object.go:493]: same here

File: ledger-core/virtual/object/object.go (hunk 1 above)
Comment (dgovorukhin) [reply ledger-core/virtual/object/object.go:478]: Fixed

File: ledger-core/virtual/object/object.go (hunk 2 above)
Comment (dgovorukhin) [reply ledger-core/virtual/object/object.go:493]: Fixed

**Pull request merged**